File: em-python/Stack.py
class Stack:
    def __init__(self):
        self.items = [None] * 10
        self.length = 0

    def push(self, item):
        self._aumentar_capacidade()
        self.items[self.length] = item
        self.length += 1

    def _aumentar_capacidade(self):
        if self.length == len(self.items):
            new_items = [None] * (len(self.items) * 2)

            for i in range(len(self.items)):
                new_items[i] = self.items[i]

            self.items = new_items

    def pop(self):
        if self.is_empty():
            return None

        position = self.length - 1
        deleted_item = self.items[position]

        self.items[position] = None
        self.length -= 1

        return deleted_item

    def peek(self):
        if self.is_empty():
            return None
        return self.items[self.length - 1]

    def is_empty(self):
        return self.length == 0

    def size(self):
        return self.length

    def clear(self):
        self.items = [None] * 10
        self.length = 0

    def __str__(self):
        result = "["
        for i in range(self.length - 1):
            result += str(self.items[i]) + ", "

        if self.length > 0:
            result += str(self.items[self.length - 1])

        result += "]"
        return result

File: em-java/src/stack/Stack.java
package stack;

public class Stack<T> {
    private T[] items;
    private int length;

   public Stack() {
       this.items = (T[]) new Object[10];
       this.length = 0;
   }

    public void push(T item) {

        this.aumentarCapacidade();
        this.items[this.length] = item;
        this.length++;
    }

    private void aumentarCapacidade() {

        if (this.length == this.items.length) {

            T[] newItemsList = (T[]) new Object[this.items.length *2];

            for(int i = 0; i < this.items.length; i++){
                newItemsList[i] = this.items[i];
            }
            this.items = newItemsList;

        }
    }

    public T pop() {
        if(this.isEmpty()){
            return null;
        }
        int POSITTION=this.length-1;

        T deletedItem=this.items[POSITTION];
        this.items[POSITTION]=null;

        this.length--;
        return deletedItem;

    }

    public boolean isEmpty() {
        return this.length == 0;
    }

    public T peek() {
        if(this.isEmpty()){
            return null;
        }
        return this.items[this.length-1];
    }

    public int size() {
        return this.length;
    }
    public void clear() {
       this.items = (T[]) new Object[10];
       this.length = 0;
    }

    @Override
    public String toString() {

        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("[");

        for (int i = 0; i < this.length-1; i++) {
            stringBuilder.append(this.items[i]);
            stringBuilder.append(", ");
        }

        if (this.length > 0) {
            stringBuilder.append(this.items[this.length-1]);
        }

        stringBuilder.append("]");
        return stringBuilder.toString();
    }
}
